File: Cpp/Number_of_Laser_Beams_in_a_Bank.cpp
class Solution {
public:
    int numberOfBeams(vector<string>& bank) {
        int prev = 0;
        int ans = 0;

        for (const string& row : bank) {
            int cur = 0;
            for (char c : row) {
                if (c == '1')
                    cur++;
            }
            if (cur > 0) {
                ans += prev * cur;
                prev = cur;
            }
        }

        return ans;
    }
};

File: Rust/Number_of_Laser_Beams_in_a_Bank.rs
impl Solution {
    pub fn number_of_beams(bank: Vec<String>) -> i32 {
        let mut prev = 0;
        let mut ans = 0;
        for row in bank {
            let cur = row.chars().filter(|&c| c == '1').count() as i32;
            if cur > 0 {
                ans += prev * cur;
                prev = cur;
            }
        }
        ans
    }
}
